tools/database/src/memtrie.rs

```diff
@@ -78,3 +90,174 @@ impl LoadMemTrieCommand {
+        let left_child_shard = child_shards[0];
+        let right_child_shard = child_shards[1];
+        println!("Left child shard: {left_child_shard}, Right child shard: {right_child_shard}");
+
+        // Create runtime and load memtrie
+        let runtime =
+            NightshadeRuntime::from_config(home, store.clone(), &near_config, epoch_manager)
+                .context("could not create the transaction runtime")?;
+        let shard_tries = runtime.get_tries();
+        println!("Loading memtrie for shard {}...", self.shard_uid);
+        shard_tries.load_memtrie(&self.shard_uid, None, false)?;
+        println!("Memtrie loaded");
+
+        // Get parent shard size
+        let size_helper = MemtrieSizeHelper::new(store.chain_store(), &shard_tries, &block);
+        let parent_size = size_helper.get_shard_trie_size(self.shard_uid)?;
+        println!("Parent shard size: {parent_size} bytes");
+
+        // Re-create epoch manager with new shard layout
+        let epoch_config_store = EpochConfigStore::for_chain_id(chain_id, None)
+            .ok_or_else(|| anyhow::anyhow!("Cannot load epoch config store"))?;
+        let mut epoch_config = (**epoch_config_store.get_config(protocol_version)).clone();
+        epoch_config.shard_layout = shard_layout;
+        let epoch_config_store =
+            EpochConfigStore::test_single_version(protocol_version, epoch_config);
+        let epoch_manager = EpochManager::new_arc_handle_from_epoch_config_store(
+            store.clone(),
+            &genesis_config,
+            epoch_config_store,
+        );
+
+        // Create resharding manager and prepare split params
+        let sender = noop().into_multi_sender();
+        let resharding_manager = ReshardingManager::new(store, epoch_manager, sender);
+        let chain_store_update = chain_store.store_update();
+        let split_params = ReshardingSplitShardParams {
+            parent_shard: self.shard_uid,
+            left_child_shard,
+            right_child_shard,
+            boundary_account: self.boundary_account.clone(),
+            resharding_block: final_head.into(),
+        };
+
+        println!("Splitting shard {} at account {}...", self.shard_uid, self.boundary_account);
+        resharding_manager.split_shard(
+            chain_store_update,
+            &block,
+            self.shard_uid,
+            shard_tries.clone(),
+            split_params,
+        )?;
+        println!("Resharding done");
+
+        let left_size = size_helper.get_shard_trie_size(left_child_shard)?;
+        let right_size = size_helper.get_shard_trie_size(right_child_shard)?;
+        println!("Left child state size: {left_size} bytes");
+        println!("Right child state size: {right_size} bytes");
+
+        println!("Removing database snapshot...");
+        snapshot.remove()?;
+        println!("Database snapshot removed");
+
+        Ok(())
+    }
+}
+
+struct MemtrieSizeHelper<'a, 'b> {
```

Comment:
nit: rename to `MemtrieSizeComputer`, `MemtrieSizeCalculator`, ... - something more related to the action done